package.json

```
@@ -10,6 +10,14 @@
     "url": "https://github.com/fabian-hiller/valibot"
   },
   "private": true,
+  "peerDependencies": {
+    "typescript": ">=5"
+  },
```

One thing to note about the TypeScript project is that [it does not follow SemVer](https://www.semver-ts.org/1-background.html#typescripts-versioning-policy), and breaking changes can be introduced in minor versions.

You might want to be more explicit about versions that are compatible with the tilde syntax:

```suggestion
  "peerDependencies": {
    "typescript": "~5.0||~5.1||~5.2||~5.3"
  },
```
Or even just a more specific range:

```suggestion
  "peerDependencies": {
    "typescript": "=5"
  },
```